zio-http/js/src/main/scala/zio/http/ZClientPlatformSpecific.scala

```diff
@@ -7,6 +7,9 @@ import zio.http.internal.FetchDriver
 
 trait ZClientPlatformSpecific {
 
+  def requiringConfig: ZLayer[ZClient.Config, Throwable, Client] =
```

Comment:
I would have ideally like to have called this something like `customized` or `configured`, but both of those are already taken by type-mismatching platform-specific overrides. Open to naming suggestions if something else is preferable.